src/bayesian_inference/emulation.py

```diff
@@ -102,7 +102,11 @@ def fit_emulator_group(config: EmulationGroupConfig) -> dict[str, Any]:
     #
     # TODO: do we want whiten=True? (NOTE: beware that inverse_transform also undoes whitening)
     scaler = sklearn_preprocessing.StandardScaler()
-    pca = sklearn_decomposition.PCA(svd_solver='full', whiten=False) # Include all PCs here, so we can access them later
+    # This adopts the sklearn convention, but then sets a max cap of 30 PCs (arbitrarily chosen) to
+    # reduce computation time.
+    max_n_components = np.min([30, *Y.shape])
```

Comment:
Perhaps we just want to move this into the config? By default (i.e. nothing specified by user), one could take all components, otherwise one can take the specified max number? 

Alternatively, and perhaps simpler, one could just have two modes: one where we only store n_pc, and another where we store all of them (i.e. there is no "max" that ever needs to be specified). 

No big preference from me. It is a minor point so we can also just proceed.

Reply:
Yeah, I guess my concern is about storage, but rather about just calculation time. My interpretation of the docs is that calculating fewer n_pc is faster, but I'm not sure if that's actually the case.

But yeah, since this was mostly a hack to speed up development, I like you're suggestion to just move it into the config. Will do that now and merge.